datajoint/condition.py

```diff
@@ -94,35 +113,59 @@ def make_condition(query_expression, condition, columns):
     from .expression import QueryExpression, Aggregation, U
 
     def prep_value(k, v):
-        """prepare value v for inclusion as a string in an SQL condition"""
-        if query_expression.heading[k].uuid:
+        """prepare SQL condition"""
+        key_match, k = translate_attribute(k)
+        if key_match["path"] is None:
+            k = f"`{k}`"
+        if (
+            query_expression.heading[key_match["attr"]].json
+            and key_match["path"] is not None
+            and isinstance(v, dict)
+        ):
+            return f"{k}='{json.dumps(v)}'"
+        if v is None:
+            return f"{k} IS NULL"
+        if query_expression.heading[key_match["attr"]].uuid:
             if not isinstance(v, uuid.UUID):
                 try:
                     v = uuid.UUID(v)
                 except (AttributeError, ValueError):
                     raise DataJointError(
                         "Badly formed UUID {v} in restriction by `{k}`".format(k=k, v=v)
                     )
-            return "X'%s'" % v.bytes.hex()
+            return f"{k}=X'{v.bytes.hex()}'"
         if isinstance(
-            v, (datetime.date, datetime.datetime, datetime.time, decimal.Decimal)
+            v,
+            (
+                datetime.date,
+                datetime.datetime,
+                datetime.time,
+                decimal.Decimal,
+                list,
+            ),
         ):
-            return '"%s"' % v
+            return f'{k}="{v}"'
         if isinstance(v, str):
-            return '"%s"' % v.replace("%", "%%")
-        return "%r" % v
+            return f'{k}="{v.replace("%", "%%")}"'
+        return f"{k}={v}"
+
+    def template(restrictions, operator=None):
```

Comment:
`operator=None` leads to an invalid template. I think it should default to `operator="AND"`.

Reply:
also, the function name `template` is not quite accurate or descriptive. It's make a condition string.

Reply:
It still works (mainly because I check `restriction` length first) but I see now a cleaner approach based on your feedback. New update coming soon.

Reply:
Perhaps `build_restriction`? or `assemble_restriction`?

Reply:
`join_conditions` also works